src/libstore/store-api.hh

```diff
@@ -372,6 +372,21 @@ public:
     void queryPathInfo(const StorePath & path,
         Callback<ref<const ValidPathInfo>> callback) noexcept;
 
+    /* Check whether the given valid path info is sufficiently well-formed
+       (e.g. hash content-address or signature) in order to be included in the
```

Comment:
"Well-formed" is a bit vague, since to me that just suggests it's syntactically valid. Maybe say "is signed by a trusted public key or is content-addressed"?